### PR TITLE
Avoid some unnecessary allocations in exec().

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

File: platform/unix/process.rs [hunk 1]
```diff
@@ -26,13 +27,11 @@ pub fn exec(command: &Command) -> io::Error {
 
     let env: Vec<_> =
         command.env.iter().map(|(key, value)| {
-            format!("{}={}",
-                    String::from_utf8(key.to_bytes().to_vec()).unwrap(),
-                    String::from_utf8(value.to_bytes().to_vec()).unwrap())
+            let entry = format!("{}={}",
+                                str::from_utf8(key.to_bytes()).unwrap(),
+                                str::from_utf8(value.to_bytes()).unwrap());
+            CString::new(entry).unwrap()
```
Comment (frewsxcv) [inline platform/unix/process.rs]: Could you call `CString::into_raw` here and then remove line 35?

Comment (Ms2ger) [inline platform/unix/process.rs]: I imagine that would be a leak.

Comment (frewsxcv) [inline platform/unix/process.rs]: Nevermind. I think I was originally thinking that `execve` would release the strings that were passed in

**Pull request merged**